ote/src/cljs/ote/app/controller/transit_visualization.cljs

```diff
@@ -470,6 +446,34 @@
                                                              earlier-date
                                                              later-date)))))))
 
+(define-event LoadServiceChangesForDateResponse [response detection-date]
+  {}
+  (let [date-filter (if (= "now" (:scope app))
+                      (time/now-iso-date-str)
+                      detection-date)
+        route-hash (js/decodeURIComponent (get-in app [:params :route]))
+        changes (future-changes date-filter (:route-changes response))
+        route (some
+                #(when
+                  (= (str/replace (:route-hash-id %) #"\s" "") route-hash)
+                  %)
+                changes)]
+    (when route-hash                                        ;;route-hash exists when you have a url where route is selected
+      (comm/get! (str "transit-visualization/" (get-in app [:params :service-id]) "/route")
+        {:params {:route-hash-id (ensure-route-hash-id route)}
+         :on-success (tuck/send-async! ->RouteCalendarDatesResponse route)}))
+    (-> app
+      (assoc :transit-visualization
+             (assoc (:transit-visualization app)
+               :service-changes-for-date-loading? false
+               :service-info (:service-info response)
+               :changes-all (sort-by :different-week-date < changes)
+               :changes-route-no-change (sorted-route-changes true changes)
+               :changes-route-filtered (sorted-route-changes false changes)
+               :gtfs-package-info (:gtfs-package-info response)
+               :route-hash-id-type (:route-hash-id-type response)))
+      (assoc-in [:transit-visualization :selected-route] route))))
```

Comment:
`(assoc[:transit-visualization :selected-route] route))))` could be moved inside the preceding `assoc`
 as plain `:selected-route route`
and threading macro removed?